pyuvsim/uvsim.py

```diff
@@ -270,37 +270,13 @@ def init_uvdata_out(uv_in, source_list_name,
 
     history += ' Npus = ' + str(mpi.Npus) + '.'
 
-    uv_obj = copy.deepcopy(uv_in)
-
-    uv_obj.set_drift()
-    uv_obj.vis_units = 'Jy'
-    uv_obj.polarization_array = np.array([-5, -6, -7, -8])
-    uv_obj.instrument = uv_obj.telescope_name
-    uv_obj.set_lsts_from_time_array()
-    uv_obj.spw_array = np.array([0])
-    if uv_obj.Nfreqs == 1:
-        uv_obj.channel_width = 1.  # Hz
-    else:
-        uv_obj.channel_width = np.diff(uv_obj.freq_array[0])[0]
-    uv_obj.set_uvws_from_antenna_positions()
-    if uv_obj.Ntimes == 1:
-        uv_obj.integration_time = np.ones_like(uv_obj.time_array, dtype=np.float64)  # Second
-    else:
-        # Note: currently only support a constant spacing of times
-        uv_obj.integration_time = (np.ones_like(uv_obj.time_array, dtype=np.float64)
-                                   * np.diff(np.unique(uv_obj.time_array))[0] * (24. * 60**2))  # Seconds
+    uv_obj = simsetup.complete_uvdata(uv_in, inplace=False)
```

Comment:
This function now just adds to the history string and runs `complete_uvdata`.... if it's okay with you, I'd like to shuffle around some of the calls in `uvsim.py` and rename this function.

Mostly, I just don't like that we now have four functions with names that suggest initializing an empty UVData object (and the docstring here suggests as much too), when it should really only be two. I'd like to rename this to "update_history" or something and briefly change how it's called.

Reply:
That seems perfectly reasonable to me!